packages/iTwinUI-react/src/core/Table/Table.test.tsx

```diff
@@ -389,6 +389,88 @@ it('should handle row clicks', async () => {
   expect(rows[2].classList).toContain('iui-selected');
   expect(onSelect).toHaveBeenCalledTimes(3);
   expect(onRowClick).toHaveBeenCalledTimes(3);
+
+  await user.keyboard('[/ControlLeft][ShiftLeft>]'); // Release Control and Press Shift (without releasing it)
+  await user.click(getByText(mockedData()[0].name)); // Perform a click with `shiftKey: true`
+
+  expect(rows[0].classList).toContain('iui-selected');
+  expect(rows[1].classList).toContain('iui-selected');
+  expect(rows[2].classList).toContain('iui-selected');
+  expect(onSelect).toHaveBeenCalledTimes(3 + (1 + 3)); // # times before + (reset + # selected rows from start to end)
+  expect(onRowClick).toHaveBeenCalledTimes(4);
+});
+
+it('should handle sub-rows shift click selection', async () => {
+  const onSelect = jest.fn();
+  const onRowClick = jest.fn();
+  const data = mockedSubRowsData();
+  const { container, getByText } = renderComponent({
+    data,
+    onSelect,
+    onRowClick,
+    isSelectable: true,
+  });
+
+  let rows = container.querySelectorAll('.iui-table-body .iui-row');
+  expect(rows.length).toBe(3);
+
+  const user = userEvent.setup();
+  await user.keyboard('[ShiftLeft>]'); // Press Shift (without releasing it)
+  await user.click(getByText(data[1].name)); // [shiftKey: true]
+
+  // By default, when no row is selected before shift click, start selecting from first row to clicked row
+  expect(rows[0].classList).toContain('iui-selected');
```

Comment:
Look like same test as in lines 393-400. Is there are reason you added both?

Reply:
Ohh, right, that's correct. They are the same. But the only difference is that here `lastSelectedRow` is `undefined`. This is because no row was selected (normal click) before the shift click ([old commit line 421](https://github.com/iTwin/iTwinUI-react/pull/774/files/da5ef1854d02894e3c653eaf158ee178b75bd210#diff-c4aa401f671f853692462c6d1ee006969f80e3d02863b8a1a8919154aa3f67d9R421)).

So this test here (417-426) is to see that is `lastSelectedRow` is `undefined`, is `lastSelectedRow` set to 0. Whereas in lines 393-400, since a row is selected before the shift click, `lastSelectedRow` is not `undefined`.

Is there a better place you suggest for adding this test? Or maybe it is okay the way it is?

Reply:
Place is good. Maybe you could add a comment explaining what you are doing? Test name does not specify lastSelectedRow status. "should handle sub-rows shift click selection" could be both either one - defined lastSelectedRow and undefined.

Reply:
**[DONE]** I tried to group similar tests together. So I now put the two shift select tests (with `lastSelectedRow`: 1. defined 2. undefined) are moved to the test: `should handle row clicks`. I added two comments `// Shift click test #1` and `#2` to make what we are testing within that test more clear.

I then left my previous code for the sub-row shift selecting test in `should handle sub-rows shift click selection` with slight improvements. Apart from testing if the appropriate checkboxes are checked, we are now also checking if the appropriate indeterminate checkboxes are indeterminate. The code is also more clean now.